### PR TITLE
Unit tests added for the Paypal controller

### PR DESCRIPTION
**Pull request by Mullefa**
cc @guardian/contributions 

Unit tests for the Paypal controller. I've had to refactor the code in this controller so the methods can be exposed for unit tests.

_Have you gone through the contributions flow for all test variants ?_ __Yes, on web__

### CODE REVIEW

File: app/controllers/PaypalController.scala
Comment (joelochlann) [inline app/controllers/PaypalController.scala:267]: As discussed - you could probably get rid of this one and just call `paypalService.executePayment` directly

**Review by joelochlann — APPROVED** [review]
Nice work Guy - all looks good from what I can tell! And moving those functions out of the controllers has cleaned up the code.

I might get a second opinion on the usefulness of testing bit of logic which are so tied to a specific controller, when we have an end-to-end test for that controller. Seems like it could be valuable as an extra safety net but I do find it quite hard to imagine a problem which would be caught by one of these tests but not by the end-to-end test. But someone with more experience than me may well be able to!

**Comment by desbo** [comment]
@joelochlann I think it's worth having these tests alongside the end-to-end ones as they cover cases that the Selenium ones won't, e.g. payment capping and errors. The selenium ones only cover the most fundamental path of completing a payment through card and PayPal, so anything extra is a bonus IMO.

@Mullefa you could probably look to merge this now that the end-to-end tests are running – I suspect the conflicts might not be much fun though 8(

**Comment by Mullefa** [comment]
yeah, gonna wait for @alexduf to merge his most recent pull request, then will resolve the conflicts 🙈

**Comment by dominickendrick** [comment]
Can this be merged ?

**Comment by Mullefa** [comment]
yes, but conflicts are going to be a real pain to resolve. I probably won't have time to do it this week.

**Pull request closed without merging**
